### PR TITLE
feat: UI for pg-cron

### PR DESCRIPTION
**Pull request by ivasilov**

### CODE REVIEW

File: apps/studio/components/interfaces/Database/CronJobs/CreateCronJobSheet.tsx [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+
+    upsertCronJob(
+      {
+        projectRef: project!.ref,
+        connectionString: project?.connectionString,
+        query,
+      },
+      {
+        onSuccess: () => {
+          if (isEditing) {
```
Comment (saltcod) [inline apps/studio/components/interfaces/Database/CronJobs/CreateCronJobSheet.tsx:188]: this need an error toast?

File: apps/studio/components/interfaces/Database/CronJobs/CronJobScheduleSection.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                      .map((s, i) => (i === 0 ? s.toLocaleLowerCase() : s))
+                      .join(' ')}
+                    .
+                  </span>
+                ) : (
+                  !fieldState.error && <span className="h-5" />
+                )}
+              </FormMessage_Shadcn_>
+            </FormItem_Shadcn_>
+            // <FormItemLayout
```
Comment (saltcod) [inline apps/studio/components/interfaces/Database/CronJobs/CronJobScheduleSection.tsx:107]: rm?

File: apps/studio/components/interfaces/Database/CronJobs/CronJobsListing.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  const pgCronExtensionInstalled = pgCronExtension?.installed_version
+
+  useEffect(() => {
+    // refetch the cron jobs afte the pgcron extension has been installed
+    if (pgCronExtensionInstalled && isError) {
+      refetch()
+    }
+  }, [isError, pgCronExtensionInstalled, refetch])
+
+  // this avoid showing loading screen when the extension is not installed. Otherwise, we';; have to wait for three
```
Comment (saltcod) [inline apps/studio/components/interfaces/Database/CronJobs/CronJobsListing.tsx:56]: ultra nit, but bunch of formatting issues in this comment

File: apps/studio/components/interfaces/Database/CronJobs/EdgeFunctionSection.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+export const EdgeFunctionSection = ({ form }: HTTPRequestFieldsProps) => {
+  const { project: selectedProject } = useProjectContext()
+  const { ref } = useParams()
+  const { data: functions, isSuccess, isLoading } = useEdgeFunctionsQuery({ projectRef: ref })
+
+  const edgeFunctions = functions ?? []
+
+  useEffect(() => {
```
Comment (saltcod) [inline apps/studio/components/interfaces/Database/CronJobs/EdgeFunctionSection.tsx:43]: this need to have functions as a dep?

File: apps/studio/components/interfaces/Database/CronJobs/HttpHeaderFieldsSection.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        {fields.map((field, index) => (
+          <div key={field.id} className="flex items-center space-x-2">
+            <FormField_Shadcn_
+              name={`${fieldName}.${index}.name`}
+              render={({ field }) => (
+                <FormItem_Shadcn_ className="flex-1">
+                  <FormControl_Shadcn_>
+                    <Input_Shadcn_
+                      {...field}
+                      size="small"
```
Comment (saltcod) [inline apps/studio/components/interfaces/Database/CronJobs/HttpHeaderFieldsSection.tsx:52]: this header dropdown relevant for the http option?

https://share.cleanshot.com/DLjtQJYf

File: apps/studio/components/interfaces/Database/CronJobs/HttpHeaderFieldsSection.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+                      className="w-full"
+                      placeholder="Header name"
+                    />
+                  </FormControl_Shadcn_>
+                  <FormMessage_Shadcn_ />
+                </FormItem_Shadcn_>
+              )}
+            />
+            <FormField_Shadcn_
```
Comment (saltcod) [inline apps/studio/components/interfaces/Database/CronJobs/HttpHeaderFieldsSection.tsx:61]: nit: chevron slightly big
https://share.cleanshot.com/W25x8c4y

**Review by saltcod — APPROVED** [review]

File: apps/studio/components/interfaces/Database/CronJobs/CreateCronJobSheet.tsx (hunk 1 above)
Comment (ivasilov) [reply apps/studio/components/interfaces/Database/CronJobs/CreateCronJobSheet.tsx:188]: no, it's to decide whether the panel is used for editing existing cron job or creating a new one.

**Pull request merged**